src/Processors/Transforms/PartialSortingTransform.h

```diff
@@ -28,15 +27,11 @@ class PartialSortingTransform : public ISimpleTransform
 
 private:
     SortDescription description;
+    SortDescriptionWithPositions description_with_positions;
     UInt64 limit;
     RowsBeforeLimitCounterPtr read_rows;
 
-    /** threshold_block is using for saving columns from previously processed block.
-      * threshold_block_columns contains pointers to columns from threshold_block which used for comparison.
-      * That's all for PartialSort optimization
-      */
-    Block threshold_block;
-    ColumnRawPtrs threshold_block_columns;
+    Columns sort_description_threshold_columns;
```

Comment:
IMO, this name isn't very descriptive. maybe just `threshold_columns`? and it will be good to have a brief comment here.

Reply:
This are columns that are only extracted as specified by sort description, it is better to be part of the name. Will add comment in separate pull request.